FetchContent error

I get an error while running the CMake configure step in a fresh copy of the repository:

```
Creating directories for 'simde-populate'
  Performing download step (git clone) for 'simde-populate'
  Cloning into 'simde-src'...
  fatal: reference is not a tree: df63f88a364da6be3963b4924c327b12f88d7748
  CMake Error at simde-subbuild/simde-populate-prefix/tmp/simde-populate-gitclone.cmake:40 (message):
    Failed to checkout tag: 'df63f88a364da6be3963b4924c327b12f88d7748'
```

Steps to reproduce:

```
git clone git@github.com:jeremyong/klein.git
cd klein
cmake -B build -S .
```

Possible fix:

Modifying [CMakeLists.txt#L35](https://github.com/jeremyong/klein/blob/49e69b8f8f4630c2c5423815f5c3332a77552140/CMakeLists.txt#L35) from `GIT_SHALLOW ON` to `GIT_SHALLOW OFF` solves the issue. I'm not sure if that's the correct approach though.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ include(FetchContent)
 FetchContent_Declare(
     simde
     GIT_REPOSITORY https://github.com/simd-everywhere/simde
-    GIT_TAG df63f88a364da6be3963b4924c327b12f88d7748
+    GIT_TAG v0.7.2
     GIT_SHALLOW ON
 )
 FetchContent_GetProperties(simde)

